command/agent/testagent.go

```diff
@@ -39,7 +38,7 @@ var TempDir = os.TempDir()
 // is removed after shutdown.
 type TestAgent struct {
 	// T is the testing object
-	T testing.T
+	T testing.TB
```

Comment:
I'm enthusiastically 👍  for removing unnecessary dependencies, but I'm not sure I know what inspired that change in this PR?

Reply:
It turns out glint updated `mitchellh/go-testing-interface`, and the update added a `Parallel()` method to that library's `testing.T`, which broke some of the tests. So this rips that out in favor of `testing.TB`, which doesn't have `Parallel()`.

I kept those changes as a separate commit, but if you want me to squash everything into one, I can do that and also add an explanation for the testing changes in the commit message

Reply:
Oh, I should have looked at the commit message! Nevermind, you're good!